src/app/code/community/Divante/VueStorefrontIndexer/Model/Index/Mapping/Review.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+use Divante_VueStorefrontIndexer_Api_MappingInterface as MappingInterface;
+use Divante_VueStorefrontIndexer_Api_Mapping_FieldInterface as FieldInterface;
+use Divante_VueStorefrontIndexer_Model_Index_Mapping_Eav_Abstract as AbstractMapping;
+use Divante_VueStorefrontIndexer_Model_Index_Mapping_Generalmapping as GeneralMapping;
+
+/**
+ * Class Divante_VueStorefrontIndexer_Model_Index_Mapping_Product
+ *
+ * @package     Divante
+ * @category    VueStoreFrontIndexer
+ * @author      Jose Castaneda <jose@qbo.tech>
+ * @copyright   Copyright (C) 2018 Divante Sp. z o.o.
+ * @license     See LICENSE_DIVANTE.txt for license details.
+ */
+class Divante_VueStorefrontIndexer_Model_Index_Mapping_Review implements MappingInterface
+{
+
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var array
+     */
+    private $properties;
+
+    /**
+     * @inheritdoc
+     */
+    public function setType($type)
+    {
+        $this->type = $type;
+    }
+
+    /**
+     * @return string
+     */
+    public function getType()
+    {
+        return $this->type;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getMappingProperties()
+    {
+        if (null === $this->properties) {
+            $attributes = $this->getAttributes();
+            $attributesMapping  = [];
+
+            foreach ($attributes as $attributeCode => $attribute) {
+                $attributesMapping = array_merge($attributesMapping, $this->getAttributeMapping($attributeCode, $attribute));
+            }
+            /**
+             * @var $generalMapping GeneralMapping
+             */
+            $generalMapping = Mage::getModel('vsf_indexer/index_mapping_generalmapping');
+            $properties = array_merge($attributesMapping, $generalMapping->getCommonProperties());
+ 
+            $mapping = ['properties' => $properties];
+
+            $mappingObject = new Varien_Object($mapping);
+            Mage::dispatchEvent('elasticsearch_review_mapping_properties', ['mapping' => $mappingObject]);
+
+            $this->properties = $mappingObject->getData();
+        }
+
+        return $this->properties;
+    }
+
+    /**
+     *
+     * @return array
+     */
+    public function getAttributes()
+    {
+       $attributes = array(
+           'product_id' => ['type' => FieldInterface::TYPE_INT],
+           'detail' => ['type' => FieldInterface::TYPE_TEXT],
+           'nickname' => ['type' => FieldInterface::TYPE_TEXT],
+           'review_entity' => ['type' => FieldInterface::TYPE_TEXT],
```

Comment:
I think `TYPE_KEYWORD` type might be better for `review_entity` and `TYPE_LONG` for `product_id`
missing `ID` field
```
"id" => ["type" => FieldInterface::TYPE_LONG]
```

Reply:
Keyword types are used for structured content like fields used for filtering, sorting or aggregations, which is not the case for this particular field ...actually Im noticing that this field is not even indexed in elasticsearch since the mysql query does not return this field. I think would be beter to remove it from mapping.

Same with product_id, it's not filterable, nor sortable so Ill leave it as LONG as you suggested.